dev/com.ibm.ws.microprofile.faulttolerance.2.0/test/src/com/ibm/ws/microprofile/faulttolerance/test/AbstractFTTest.java

```diff
@@ -25,7 +29,9 @@
     @Before
     public void before() {
         System.setProperty(FTConstants.JSE_FLAG, "true");
-        FaultToleranceProviderResolver.setInstance(new ProviderResolverImpl20());
+        ProviderResolverImpl20 providerResolver = new ProviderResolverImpl20();
+        setField(ProviderResolverImpl.class, "contextService", providerResolver, new DummyContextService());
```

Comment:
Wouldn't setters and getters on ProviderResolverImpl be better rather than trying to access protected fields like this? I think those fields should be private anyway

Reply:
I don't think they're the fields you're thinking of.

These fields are annotated `@Reference` and populated with services provided by DS.

I think setting the references by reflection is an appropriate approach for a unit test. I'd prefer to avoid adding further boilerplate to the implementation where it isn't necessary.